internal/http.go

```diff
@@ -0,0 +1,206 @@
+package internal
+
+import (
+	"archive/tar"
+	"compress/gzip"
+	"context"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"net"
+	"net/http"
+	"os"
+	"os/signal"
+	"path"
+	"strings"
+	"syscall"
+	"time"
+
+	"github.com/jackc/pgx/v4"
+	ctrl "sigs.k8s.io/controller-runtime"
+)
+
+const (
+	postgresURL    = "host=postgresql.test-project.svc  port=5432 user=curator password=M5rBgWkN8LfjeyI8 dbname=curatordb sslmode=disable"
+	httpServerPort = ":5000"
+)
+
+var (
+	reportLog = ctrl.Log.WithName("report")
+)
+
+type Response struct {
+	Message string `json:"message"`
+}
+
+func InitializeRouter() {
+	ctx, cancel := context.WithCancel(context.Background())
+	http.HandleFunc("/download", downloadReport)
+
+	httpServer := &http.Server{
+		Addr:        httpServerPort,
+		BaseContext: func(_ net.Listener) context.Context { return ctx },
+	}
+
+	reportLog.Info("starting http server on port 5000\n")
+	if err := httpServer.ListenAndServe(); err != nil {
+		log.Fatal(err)
+	}
+	httpServer.RegisterOnShutdown(cancel)
+
+	signalChan := make(chan os.Signal, 1)
+
+	signal.Notify(
+		signalChan,
+		syscall.SIGHUP,
+		syscall.SIGINT,
+		syscall.SIGQUIT,
+	)
```

Comment:
This kind of signal handling needs to be build up in the main.go imo. Maybe we can wrap this kind of signal handling as a helper function? See https://github.com/operator-framework/operator-marketplace/blob/master/pkg/signals/signals.go for what we've done in the past.